thrust/thrust/system/cuda/detail/core/triple_chevron_launch.h

```diff
@@ -84,42 +84,38 @@ struct _CCCL_VISIBILITY_HIDDEN triple_chevron
   size_t _CCCL_DEVICE argument_pack_size(size_t size, Args const&...) const
   {
     // TODO(bgruber): replace by fold over comma in C++17 (make sure order of evaluation is left to right!)
-    int dummy[] = {(size += align_up<Args>(size) + sizeof(Args), 0)...};
-    (void) dummy;
+    int dummy[] = {(size = align_up<Args>(size) + sizeof(Args), 0)...};
+    static_cast<void>(dummy);
     return size;
   }
 
   template <class Arg>
-  void _CCCL_DEVICE copy_arg(char* buffer, size_t& offset, Arg arg) const
+  void _CCCL_DEVICE copy_arg(void* buffer, size_t& offset, const Arg& arg) const
   {
     // TODO(bgruber): we should make sure that we can actually byte-wise copy Arg, but this fails with some tests
     // static_assert(::cuda::std::is_trivially_copyable<Arg>::value, "");
-
     offset = align_up<Arg>(offset);
-    for (int i = 0; i != sizeof(Arg); ++i)
-    {
-      buffer[offset + i] = reinterpret_cast<const char*>(&arg)[i];
-    }
+    ::memcpy(buffer, static_cast<const void*>(&arg), sizeof(arg));
```

Comment:
You may get a compiler warning here if `Arg` is not trivially copyable. I assume this is why the original implementation did not use `memcpy`.

Reply:
So this was no problem after all? Great!

Reply:
yes, `static_cast<const void*>` avoid the issue